src/main/java/org/opensearch/flowframework/indices/GlobalContextHandler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.indices;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.workflow.CreateIndexStep;
+
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX_MAPPING;
+import static org.opensearch.flowframework.workflow.CreateIndexStep.getIndexMappings;
+
+/**
+ * A handler for global context related operations
+ */
+public class GlobalContextHandler {
+    private static final Logger logger = LogManager.getLogger(GlobalContextHandler.class);
+    private final Client client;
+    private final CreateIndexStep createIndexStep;
+
+    /**
+     * constructor
+     * @param client the open search client
+     * @param createIndexStep create index step
+     */
+    public GlobalContextHandler(Client client, CreateIndexStep createIndexStep) {
+        this.client = client;
+        this.createIndexStep = createIndexStep;
+    }
+
+    /**
+     * Get global-context index mapping
+     * @return global-context index mapping
+     * @throws IOException if mapping file cannot be read correctly
+     */
+    public static String getGlobalContextMappings() throws IOException {
+        return getIndexMappings(GLOBAL_CONTEXT_INDEX_MAPPING);
+    }
+
+    private void initGlobalContextIndexIfAbsent(ActionListener<Boolean> listener) {
+        createIndexStep.initIndexIfAbsent(FlowFrameworkIndex.GLOBAL_CONTEXT, listener);
```

Comment:
Curious why the functionality to create our system indices is within the `CreateIndexStep`, I would assume the workflow steps are only used during template execution to set up the infrastructure of the builder's application. Perhaps lets have a separate index utility class that both the `CreateIndexStep` and `GlobalContextHandler` can use.

Reply:
Maybe something like the [`IndexManagement`](https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/timeseries/indices/IndexManagement.java) class in AD : we can keep all the logic we need for managing the lifecycle of our plugin's system indices in a single class.

Reply:
I had a separate class like `IndexManagement` in my previous pr. But after @owaiskazi19 `CreateIndexStep` pr, Owais suggested me to move methods in the `IndexManagement` file under this file. I don't have a strong preference, but I closed my previous pr for this refactoring reason

Reply:
Additionally, we'll want to register the global context index with OpenSearch using this [`SystemIndexPlugin`](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/plugins/SystemIndexPlugin.java). We'll have to make the `FlowFrameworkPlugin` class extend this.

More information on how to correctly handle system indices is called out [here](https://github.com/opensearch-project/job-scheduler/issues/466)